chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/PendingSurveillanceManagerImpl.java

```diff
@@ -0,0 +1,728 @@
+            alreadyDeletedEx.setEndDate(pendingSurv.getEndDate());
+
+            try {
+                UserDTO lastModifiedUser = userDao.getById(pendingSurv.getLastModifiedUser());
+                if (lastModifiedUser != null) {
+                    Contact contact = new Contact();
+                    contact.setFullName(lastModifiedUser.getFullName());
+                    contact.setFriendlyName(lastModifiedUser.getFriendlyName());
+                    contact.setEmail(lastModifiedUser.getEmail());
+                    contact.setPhoneNumber(lastModifiedUser.getPhoneNumber());
+                    contact.setTitle(lastModifiedUser.getTitle());
+                    alreadyDeletedEx.setContact(contact);
+                } else {
+                    alreadyDeletedEx.setContact(null);
+                }
+            } catch (final UserRetrievalException ex) {
+                alreadyDeletedEx.setContact(null);
+            }
+            throw alreadyDeletedEx;
+        }
+        //If pendingSurv were null, we would have gotten an NPE by this point
+        //return pendingSurv != null;
+        return true;
+    }
+
+    private Integer getSurveillanceRecordThreshold() {
+        String surveillanceThresholdToProcessAsJobStr = env.getProperty("surveillanceThresholdToProcessAsJob").trim();
+        Integer surveillanceThresholdToProcessAsJob = SURV_THRESHOLD_DEFAULT;
+        try {
+            surveillanceThresholdToProcessAsJob = Integer.parseInt(surveillanceThresholdToProcessAsJobStr);
+        } catch (final NumberFormatException ex) {
+            LOGGER.error(
+                    "Could not format " + surveillanceThresholdToProcessAsJobStr + " as an integer. Defaulting to"
+                            + " 50 instead.");
+        }
+        return surveillanceThresholdToProcessAsJob;
+    }
+
+    private void validate(final Surveillance surveillance) {
+        validator.validate(surveillance, false);
+    }
+
+    private void checkSurveillanceAuthority(final Surveillance surv) throws SurveillanceAuthorityAccessDeniedException {
+        Boolean hasOncAdmin = Util.isUserRoleAdmin();
+        Boolean hasAcbAdmin = Util.isUserRoleAcbAdmin();
+        if (StringUtils.isEmpty(surv.getAuthority())) {
+            // If user has ROLE_ADMIN and ROLE_ACB
+            // return 403
+            if (hasOncAdmin && hasAcbAdmin) {
+                String errorMsg = "Surveillance cannot be created by user having " + Authority.ROLE_ADMIN + " and "
+                        + Authority.ROLE_ACB;
+                LOGGER.error(errorMsg);
+                throw new SurveillanceAuthorityAccessDeniedException(errorMsg);
+            }
+        } else {
+            // Cannot have surveillance authority as ROLE_ADMIN for user lacking
+            // ROLE_ADMIN
+            if (surv.getAuthority().equalsIgnoreCase(Authority.ROLE_ADMIN) && !hasOncAdmin) {
+                String errorMsg = "User must have authority " + Authority.ROLE_ADMIN;
```

Comment:
Why is the check against having ROLE_ONC, but the error says they must have ROLE_ADMIN?

I'm really confused by this method